src/lib/support/Span.h

```diff
@@ -0,0 +1,46 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <cstdlib>
+
+namespace chip {
+
+/**
+ * @brief A wrapper class for holding objects and its length, without the ownership of it.
+ * We can use C++20 std::span once we support it, the begin() and size() come from C++20 std::span.
```

Comment:
There is no `begin()`.  Did you mean to rename `data()` to `begin()`?

Reply:
Done, thanks